source/ptreferential/ptreferential.cpp

```diff
@@ -102,12 +102,20 @@ template<class T>
 WhereWrapper<T> build_clause(std::vector<Filter> filters) {
     WhereWrapper<T> wh(new BaseWhere<T>());
```

Comment:
minor note(even shouldn't be in this PR): with `new`, it's not exception safe. `make_shared` should be used